server/src/test/java/org/apache/cassandra/sidecar/restore/RestoreJobDiscovererTest.java

```diff
@@ -379,6 +385,59 @@ void testAdjustRecencyDays()
         .isEqualTo(10);
     }
 
+    @Test
+    void testSkipNotOwnedRestoreToLocalDatacenterOnlyJob()
+    {
+        // Create a restore job that restores to dc2 only. Meanwhile, discoverer runs in dc1.
+        UUID jobId = UUIDs.timeBased();
+        when(mockJobAccessor.findAllRecent(anyLong(), anyInt()))
+        .thenReturn(Collections.singletonList(RestoreJob.builder()
+                                                        .createdAt(RestoreJob.toLocalDate(jobId))
+                                                        .jobId(jobId)
+                                                        .jobAgent("agent")
+                                                        .jobStatus(RestoreJobStatus.CREATED)
+                                                        .expireAt(new Date(System.currentTimeMillis() + 10000L))
+                                                        .consistencyLevel(ConsistencyLevel.LOCAL_QUORUM)
+                                                        .localDatacenter("dc2")
+                                                        .shouldRestoreToLocalDatacenterOnly(true)
+                                                        .build()));
+        executeBlocking();
+        verify(mockRangeAccessor, never()).create(any());
+    }
+
+    @Test
+    void testRestoreToLocalDatacenterOnlyJobIsOnHoldWhenLocalDatacenterInUndetermined()
```

Comment:
Is this what you meant?
```suggestion
    void testRestoreToLocalDatacenterOnlyJobIsOnHoldWhenLocalDatacenterIsUndetermined()
```